src/SFA.DAS.ProviderApprenticeshipsService.Web/Orchestrators/BulkUploadOrchestrator.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using MediatR;
+using NLog;
+
+using SFA.DAS.ProviderApprenticeshipsService.Application.Queries.GetCommitment;
+using SFA.DAS.ProviderApprenticeshipsService.Application.Queries.GetFrameworks;
+using SFA.DAS.ProviderApprenticeshipsService.Application.Queries.GetStandards;
+using SFA.DAS.ProviderApprenticeshipsService.Domain;
+using SFA.DAS.ProviderApprenticeshipsService.Domain.Interfaces;
+using SFA.DAS.ProviderApprenticeshipsService.Web.Models;
+using SFA.DAS.ProviderApprenticeshipsService.Web.Models.BulkUpload;
+using SFA.DAS.ProviderApprenticeshipsService.Web.Orchestrators.BulkUpload;
+
+namespace SFA.DAS.ProviderApprenticeshipsService.Web.Orchestrators
+{
+    public class BulkUploadOrchestrator
+    {
+        private readonly IMediator _mediator;
+
+        private readonly BulkUploader _bulkUploader;
+
+        private readonly IHashingService _hashingService;
+
+        private readonly ILogger _logger = LogManager.GetCurrentClassLogger();
+
+        public BulkUploadOrchestrator(IMediator mediator, BulkUploader bulkUploader, IHashingService hashingService)
+        {
+            _mediator = mediator;
+            _bulkUploader = bulkUploader;
+            _hashingService = hashingService;
+        }
+
+        public async Task<BulkUploadResult> UploadFile(UploadApprenticeshipsViewModel uploadApprenticeshipsViewModel)
```

Comment:
I'd probably suggest trying to move most of this code into a separate class. Maybe the BulkUploader. Is it that case that we just need something to return a list of validation error or an empty list give file contents?